product_expiry_removal_required/models/stock_production_lot.py

```diff
@@ -0,0 +1,18 @@
+# Copyright NuoBiT - Eric Antones <eantones@nuobit.com>
+# Copyright NuoBiT 2025 - Bijaya Kumal <bkumal@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class StockProductionLot(models.Model):
```

Comment:
Rename also this
```suggestion
class StockLot(models.Model):
```
And also the filename `stock_lot.py` instead of` stock_production_lot.py`
And the views, `stock_production_lot_views.xml` to `stock_lot_views.xml`